projects/batfish-client/src/main/java/org/batfish/client/Client.java

```diff
@@ -1638,11 +1658,31 @@ private boolean initOrAddAnalysis(
     String questionsPathStr = parameters.get(1);
 
     Map<String, String> questionMap = new TreeMap<>();
+    LoadQuestionAnswerElement ae = new LoadQuestionAnswerElement();
 
-    if (!loadQuestions(null, questionsPathStr, questionMap)) {
+    Multimap<String, String> analysisQuestions = HashMultimap.create();
+    try {
+      loadQuestionsFromDir(questionsPathStr, analysisQuestions);
+    } catch (Exception e) {
       return false;
     }
 
+    Answer answer = new Answer();
+    answer.addAnswerElement(ae);
+    mergeQuestions(analysisQuestions, questionMap, ae);
+    ObjectMapper mapper = new BatfishObjectMapper(getCurrentClassLoader());
+    String answerStringToPrint;
+    try {
+      answerStringToPrint = mapper.writeValueAsString(answer);
+    } catch (JsonProcessingException e) {
+      throw new BatfishException("Could not write answer element as string", e);
+    }
+    if (_settings.getPrettyPrintAnswers()) {
```

Comment:
it looks to me like this logic is messed up because what we want is to get the pretty print only if `outWriter` is null. And you dropped that check.

Reply:
it also seems weird to me that we do L1676 unconditionally.

Reply:
discussed (1) in person, (2) is resolved